pkg/detector/eventfilter_test.go

```diff
@@ -0,0 +1,274 @@
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "nginx_v1"},
+					},
+				},
+				Status: appsv1.DeploymentStatus{Replicas: 3, UpdatedReplicas: 3, ReadyReplicas: 0},
+			},
+			newObj: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123498",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}, {}},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "nginx_v2"},
+					},
+				},
+				Status: appsv1.DeploymentStatus{Replicas: 3, UpdatedReplicas: 1, ReadyReplicas: 2},
+			},
+			wantChange: true,
+		},
+		{
+			name: "Only user defined fields changed(ConfigMap)",
+			oldObj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Immutable:  nil,
+				Data:       map[string]string{"color": "red"},
+				BinaryData: nil,
+			},
+			newObj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Immutable:  nil,
+				Data:       map[string]string{"color": "green"},
+				BinaryData: nil,
+			},
+			wantChange: true,
+		},
+		{
+			name: "Only status fields changed(ConfigMap)",
+			oldObj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Immutable:  nil,
+				Data:       map[string]string{"color": "red"},
+				BinaryData: nil,
+			},
+			newObj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123498",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}, {}},
+				},
+				Immutable:  nil,
+				Data:       map[string]string{"color": "red"},
+				BinaryData: nil,
+			},
+			wantChange: false,
+		},
+		{
+			name: "Both user defined fields and status fields changed(ConfigMap)",
+			oldObj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Immutable:  nil,
+				Data:       map[string]string{"color": "red"},
+				BinaryData: nil,
+			},
+			newObj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123498",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}, {}},
+				},
+				Immutable:  nil,
+				Data:       map[string]string{"color": "green"},
+				BinaryData: nil,
+			},
+			wantChange: true,
+		},
+		{
+			name: "Only user defined fields changed(CRD_WorkLoad)",
+			oldObj: &workloadv1alpha1.Workload{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Spec: workloadv1alpha1.WorkloadSpec{
+					Replicas: ptrInt32(3),
```

Comment:
```suggestion
					Replicas: pointer.Int32(3),
```
import `"k8s.io/utils/pointer"`